src/ILVerify/src/ILImporter.StackValue.cs

```diff
@@ -216,7 +216,7 @@ bool IsAssignable(StackValue src, StackValue dst)
                 if (src.Type == null)
                     return true;
 
-                return IsAssignable(src.Type, dst.Type);
+                return CastingHelper.CanCastTo(src.Type, dst.Type);
```

Comment:
Change `IsAssignable(TypeDesc src, TypeDesc dst)` to call `CastingHelper.CanCastTo` as well while you are on it?